GenericsHomework/GenericsHomework/Node.cs

```diff
@@ -0,0 +1,97 @@
+﻿namespace GenericsHomework
+{
+    public class Node<T> where T : notnull //maybe make this nullable...

```

Comment:
```suggestion
    public class Node<T> 
```
You shouldn't really need to add this on at all.